vesting/lib.rs

```diff
@@ -192,17 +223,32 @@ mod vesting_contract {
             &self,
             total_allocation: Balance,
             current_time: Timestamp,
-        ) -> Balance {
+        ) -> Result<Balance> {
             if current_time < self.start_time() {
-                return 0;
-            } else if current_time >= self.end_time() {
-                return total_allocation;
+                Ok(0)
+            } else if current_time >= self.end_time()? {
+                Ok(total_allocation)
             } else {
-                return (total_allocation
-                    * (current_time - self.start_time()) as Balance)
-                    / self.duration_time() as Balance;
+
+                let time_elapsed = current_time
+                    .checked_sub(self.start_time())
+                    .ok_or(Error::TimestampError)?;
+
+                // This represents the portion of the total allocation 
+                // that has vested based on the time elapsed.
+                let vested_portion = total_allocation
+                    .checked_mul(time_elapsed as u128)
+                    .ok_or(Error::Overflow)?;
+
+                let total_vesting_duration = self.duration_time() as u128;
+
+                let vested_balance = vested_portion
+                    .checked_div(total_vesting_duration)
```

Comment:
Duration time should never be 0, this should be enforced during initialisation

Reply:
> Duration time should never be 0, this should be enforced during initialisation

done